Sources/SAPCAI/UI/Modifiers/Modifiers.swift

```diff
@@ -92,3 +93,42 @@ extension View {
         self.modifier(TextWidthModifier(geometry: geometry, isRegular: isRegular, isBot: isBot))
     }
 }
+
+public struct ChangeObserver<V: Equatable>: ViewModifier {
+    public init(newValue: V, action: @escaping (V) -> Void) {
+        self.newValue = newValue
+        self.newAction = action
+    }
+
+    private typealias Action = (V) -> Void
+
+    private let newValue: V
+    private let newAction: Action
+
+    @State private var state: (V, Action)?
+
+    public func body(content: Content) -> some View {
+        if #available(iOS 14, *) {
+            assertionFailure("Please don't use this ViewModifer directly and use the `onChange(of:perform:)` modifier instead.")
+        }
+        return content
+            .onAppear()
+            .onReceive(Just(self.newValue)) { newValue in
+                if let (currentValue, action) = state, newValue != currentValue {
+                    action(newValue)
+                }
+                state = (newValue, newAction)
+            }
+    }
+}
+
+public extension View {
+    @_disfavoredOverload
```

Comment:
why is this needed? doesn't the use of `@_disfavoredOverload` means that the iOS 14  SwiftUI instance method `func onChange<V>(of value: V, perform action: @escaping (V) -> Void) -> some View where V : Equatable` is the preferred one when available? Hence I am wondering if the `if #available` block inside is necessary. I am not sure so just asking...

Reply:
I have update drag gesture for carousel page, so I this modifier is unnecessary. I will remove this.